plugin/core/views.py

```diff
@@ -176,20 +176,67 @@ def text_document_range_formatting(view: sublime.View, region: sublime.Region) -
     })
 
 
-def minihtml(view: sublime.View, content: Union[str, dict]) -> str:
-    """ Content can be a string or MarkupContent. """
+def minihtml(view: sublime.View, content: Union[str, dict, list], prefer_plain_text: bool) -> str:
+    """
+    Formats provided input into markup accepted by minihtml.
+
+    Content can be in one of those formats:
+     - string: treated as plain text
+     - MarkedString: string or { language: string; value: string }
+     - MarkedString[]
+     - MarkupContent: { kind: MarkupKind, value: string }
+
+    Since we can't distinguish between plain text string and a MarkedString in a string form, we
+    need to be told explicitly which one to use for string input.
+
+    :param view
+    :param content
+    :param prefer_plain_text: Whether to treat *string* input as plain text (True) or markdown (False).
+
+    :returns: Formatted string
+    """
+    is_plain_text = True
     if isinstance(content, str):
-        return text2html(content)
-    elif isinstance(content, dict):
-        value = content.get("value") or ""
+        # plain text string or MarkedString
+        is_plain_text = prefer_plain_text
+        # return text2html(content) if prefer_plain_text else mdpopups.md2html(view, content)
+    if isinstance(content, list):
+        # MarkedString[]
+        formatted = []
+        for item in content:
+            value = ""
+            language = None
+            if isinstance(item, str):
+                value = item
+            else:
+                value = item.get("value") or ""
+                language = item.get("language")
+
+            if language:
+                formatted.append("```{}\n{}\n```\n".format(language, value))
+            else:
+                formatted.append(value)
+
+        is_plain_text = False
+        content = "\n".join(formatted)
+    if isinstance(content, dict):
+        # MarkupContent or MarkedString (dict)
+        language = content.get("language")
         kind = content.get("kind")
-        if kind == "markdown":
-            return mdpopups.md2html(view, value)
-        else:
-            # must be plaintext
-            return text2html(value)
+        value = content.get("value") or ""
+        if kind:
+            # MarkupContent
+            is_plain_text = kind != "markdown"
+            content = value
+        if language:
+            # MarkedString (dict)
+            is_plain_text = False
+            content = "```{}\n{}\n```\n".format(language, value)
+    if is_plain_text:
+        return text2html(content)
     else:
-        return ''
+        frontmatter_config = mdpopups.format_frontmatter({'allow_code_wrap': True})
```

Comment:
We can revert if it proves problematic but we are now always setting `allow_code_wrap` for markup content.